src/server.rs

```diff
@@ -1,24 +1,41 @@
 use crate::failure::Error;
-use crate::rocket;
+use crate::rocket::config::{Config, Environment, LoggingLevel};
+use crate::rocket::response::content;
+use crate::rocket::Rocket;
+use crate::rocket::State;
+
+use crate::logger::Logger;
 use crate::settings::Settings;
 
 pub struct Server {}
 
 impl Server {
-    pub fn new(settings: &Settings) -> Result<rocket::Rocket, Error> {
-        let env = settings.environment.clone();
-        let rocket_env = env.parse::<rocket::config::Environment>().unwrap();
+    pub fn new(settings: &Settings, logger: Logger) -> Result<Rocket, Error> {
+        logger.debug("Setting up server environment");
+
+        let environment = settings.environment.clone();
+        let rocket_env = environment.parse::<Environment>().unwrap();
 
-        let config = rocket::config::Config::build(rocket_env)
+        logger.debug("Setting up server address and port");
+        let config = Config::build(rocket_env)
             .address(&settings.server.address)
             .port(settings.server.port)
-            .finalize()?;
+            .log_level(LoggingLevel::Off)
+            .finalize()
+            .unwrap();
```

Comment:
What's the difference between `.finalize()?` and `.finalize().unwrap()` ?

Reply:
Mhh, is actually better to use `?`, `unwrap` panics if there was an error, the `?` is like a throw that should be handled by the parent function.

[Here](https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/the-question-mark-operator-for-easier-error-handling.html) is a bit of information.

I should start migrating to `?` whenever possible, good catch :)